docs/api-reference/edge-runtime.md

```diff
@@ -136,6 +136,25 @@ The following JavaScript language features are disabled, and **will not work:**
 
 - `eval`: Evaluates JavaScript code represented as a string
 - `new Function(evalString)`: Creates a new function with the code provided as an argument
+- `WebAssembly.compile`
+- `WebAssembly.instantiate` with [a buffer parameter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/WebAssembly/instantiate#primary_overload_%E2%80%94_taking_wasm_binary_code)
+
+In rare cases, your code could contain (or import) some dynamic code evaluation statements which _can not be reached at runtime_ and which can not be removed by threeshaking.
+You can relax the check to allow specific files with your Middleware or Edge API Route exported configuration:
+
+```javascript
+export const config = {
+  runtime: 'experimental-edge', // for Edge API Routes only
+  allowDynamic: [
+    '/lib/utilities.js', // allows a single file
+    '/node_modules/function-bind/**', // use a glob to allow anything in the function-bind 3rd party module
+  ],
+}
```

Comment:
I have two questions:

- Is it possible to exclude the file defining `config` itself?
- Do we need to specify `allocDynamic` in every api routes even if they have the same rule?

Reply:
> Is it possible to exclude the file defining config itself?

Yes it is, as long as you provide a matching pattern.
For example:
```js
// /middleware.js
import { NextResponse } from 'next/server'
import { hasUnusedDynamic } from './lib'

export default async function middleware() {
  const response = NextResponse.next()
  if ((() => false)()) {
    eval('100')
  }
  hasUnusedDynamic()
  response.headers.append('x-from-middleware', true)
  return response
}

export const config = {
  allowDynamic: ['/lib/**', '/middleware'],
}
```

> Do we need to specify allowDynamic in every api routes even if they have the same rule?

Yes we do, which is a bit annoying. The alternative would be to support `allowDynamic` in next.config.js. 
It was considered in the RFC, but left aside, and I can't remember why 😢

Reply:
Thanks for the clarification! Regarding the second question, I think it can be done later in a separate PR 😄